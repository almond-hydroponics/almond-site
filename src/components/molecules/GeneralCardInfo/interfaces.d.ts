import { ReactNode } from 'react';

export interface GeneralCardInfoProps {
	mainHeader: string;
	subHeader?: string;
	actionItem?: ReactNode;
	icon?: ReactNode | string;
}
