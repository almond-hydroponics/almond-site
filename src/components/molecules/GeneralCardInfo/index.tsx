import { ButtonBase, Grid, Typography, Box } from '@mui/material';
import { GeneralCardInfoProps } from './interfaces';
import { useTheme } from '@mui/material/styles';

const GeneralCardInfo = ({
	mainHeader,
	subHeader,
	actionItem,
	icon,
}: GeneralCardInfoProps): JSX.Element => {
	const theme = useTheme();

	return (
		<Grid
			xs={12}
			sx={{
				height: 'fit-content',
				border: `1px solid ${theme.palette.divider}`,
				borderRadius: 2,
				backgroundColor: theme.palette.background.paper,
				marginBottom: 2,
				[theme.breakpoints.up('sm')]: {
					marginRight: 1,
				},
			}}
		>
			<Box padding={2} display={'flex'} alignItems={'center'}>
				<Box marginRight={2}>
					<ButtonBase
						sx={{
							padding: 1,
							borderRadius: 2,
							backgroundColor: 'rgba(25, 103, 210, 0.11)',
							color: '#2573b5',
						}}
					>
						{icon}
					</ButtonBase>
				</Box>
				<Box
					display={'flex'}
					flexDirection={{ xs: 'column', sm: 'row' }}
					flex={'1 1 100%'}
					justifyContent={{ sm: 'space-between' }}
					alignItems={{ sm: 'center' }}
				>
					<Box marginBottom={{ xs: 1, sm: 0 }}>
						<Typography color="primary" variant={'subtitle1'} fontWeight={700}>
							{mainHeader}
						</Typography>
						<Typography color={'text.secondary'} fontSize={14}>
							{subHeader}
						</Typography>
					</Box>
				</Box>
				<Box marginLeft={2}>{actionItem}</Box>
			</Box>
		</Grid>
	);
	// return (
	// 	<Grid
	// 		item
	// 		xs={12}
	// 		sx={{
	// 			paddingBottom: 0,
	// 			[theme.breakpoints.down('sm')]: {
	// 				paddingTop: '0 !important',
	// 				marginBottom: 8,
	// 			},
	// 		}}
	// 	>
	// 		<Card
	// 			sx={{ borderRadius: 2 }}
	// 			variant="outlined"
	// 			data-testid="general-card-info"
	// 		>
	// 			<CardContent
	// 				sx={{ flex: '1 0 auto', flexDirection: 'row', width: '100%' }}
	// 			>
	// 				<Grid
	// 					item
	// 					container
	// 					direction="row"
	// 					justifyContent="space-between"
	// 					alignItems="center"
	// 					// spacing={2}
	// 					style={{ display: 'flex' }}
	// 					xs={12}
	// 				>
	// 					{icon && (
	// 						<Grid
	// 							item
	// 							container
	// 							xs={2}
	// 							direction="row"
	// 							alignItems="center"
	// 							justifyContent="center"
	// 						>
	// 							<ButtonBase
	// 								sx={{
	// 									padding: 1,
	// 									borderRadius: 2,
	// 									backgroundColor: 'rgba(25, 103, 210, 0.11)',
	// 									color: '#2573b5',
	// 								}}
	// 							>
	// 								{icon}
	// 							</ButtonBase>
	// 						</Grid>
	// 					)}
	// 					<Grid item container xs={icon ? 8 : 12}>
	// 						<Stack spacing={0}>
	// 							<Typography variant="h6" color="primary" data-testid="header">
	// 								{mainHeader}
	// 							</Typography>
	// 							<Typography
	// 								variant={isMd ? 'subtitle1' : 'body2'}
	// 								data-testid="sub-header"
	// 								fontSize={14}
	// 							>
	// 								{subHeader}
	// 							</Typography>
	// 						</Stack>
	// 					</Grid>
	// 					<Grid
	// 						item
	// 						container
	// 						xs={2}
	// 						direction="row"
	// 						alignItems="center"
	// 						justifyContent="flex-end"
	// 					>
	// 						{actionItem}
	// 					</Grid>
	// 				</Grid>
	// 			</CardContent>
	// 		</Card>
	// 	</Grid>
	// );
};

export default GeneralCardInfo;
