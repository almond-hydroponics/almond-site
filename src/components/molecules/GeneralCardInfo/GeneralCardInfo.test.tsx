// react libraries
import { render, screen } from '@testing-library/react';

// component
import GeneralCardInfo from './index';

describe('GeneralCardInfo component', () => {
	const props = {
		mainHeader: 'mainHeader',
		subHeader: 'subHeader',
	};

	it('should render correctly', () => {
		const { asFragment } = render(<GeneralCardInfo {...props} />);
		expect(asFragment()).toMatchSnapshot();

		const elemHeader = screen.getByTestId('header');
		expect(elemHeader.innerHTML).toBe('mainHeader');

		const elemDetails = screen.getByTestId('sub-header');
		expect(elemDetails.innerHTML).toBe('subHeader');
	});
});
