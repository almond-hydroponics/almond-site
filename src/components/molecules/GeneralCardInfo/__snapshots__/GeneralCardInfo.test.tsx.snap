// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralCardInfo component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-mainCard-8 MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-1 makeStyles-cardPaper-5 MuiPaper-outlined MuiPaper-rounded"
      data-testid="general-card-info"
    >
      <div
        class="MuiCardContent-root makeStyles-content-3"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between MuiGrid-grid-xs-12"
          style="display: flex;"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-2"
          >
            <button
              class="MuiButtonBase-root makeStyles-iconBase-9"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-space-between MuiGrid-grid-xs-7"
            style="display: flex; width: 100%;"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary"
              data-testid="header"
            >
              mainHeader
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1"
              data-testid="sub-header"
            >
              subHeader
            </h6>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-3"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
